analysis/cross_partition_combiners.py

```diff
@@ -83,28 +83,34 @@ def _sum_metrics_to_metric_utility(
         sum_metrics, keep_prob=partition_keep_probability)
     relative_error = absolute_error.to_relative(sum_metrics.sum)
 
-    return metrics.MetricUtility(
-        metric=dp_metric,
-        num_dataset_partitions=1,
-        num_non_public_partitions=0,  # todo(dvadym): to implement it.
-        num_empty_partitions=1 if is_empty_public else 0,
-        noise_std=sum_metrics.std_noise,
-        noise_kind=sum_metrics.noise_kind,
-        ratio_data_dropped=data_dropped,
-        absolute_error=absolute_error,
-        relative_error=relative_error)
-
-
-def _partition_selection_per_to_cross_partition(
-        prob_keep: float) -> metrics.PrivatePartitionSelectionMetrics:
-    """Creates cross-partition partition selection metrics from keep probability for 1 partition."""
-    return metrics.PrivatePartitionSelectionMetrics(
-        strategy=None,
-        num_partitions=1,
-        dropped_partitions=metrics.MeanVariance(mean=prob_keep,
-                                                var=prob_keep *
-                                                (1 - prob_keep)),
-        ratio_dropped_data=0)  # todo(dvadym): implement ratio_dropped_data
+    return metrics.MetricUtility(metric=dp_metric,
+                                 noise_std=sum_metrics.std_noise,
+                                 noise_kind=sum_metrics.noise_kind,
+                                 ratio_data_dropped=data_dropped,
+                                 absolute_error=absolute_error,
+                                 relative_error=relative_error)
+
+
+def _partition_metrics_public_partitions(
```

Comment:
Optional: Maybe rename to `_to_partition_metrics()` instead and handle both cases (private and public) inside the same function?

Reply:
I thought about that, but that function is going to be pretty complicated, that's why I split in 2 functions.